File: Data Structures & Algorithms/binary-tree-diameter/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int d = 0;
    public int diameterOfBinaryTree(TreeNode root) {
        heightOfNode(root);
        return d;
    }
    public int heightOfNode(TreeNode root){
        if(root == null){return 0;}

        int right = heightOfNode(root.right);
        int left = heightOfNode(root.left);

        d = Math.max(d,left+right);

        return Math.max(right,left)+1;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0,right = heights.size()-1,area,m=0;
        while(left<right){
            area = (right - left)*min(heights[right],heights[left]);
            if(heights[right]>heights[left]){
                left++;
            }else{
                right--;
            }
            m = max(m,area);
        }
        return m;
    }
};
